### ISSUE TITLE
Ruby builds are not reproducible

### ISSUE THREAD

**Issue by Mindavi** [NixOS/nixpkgs#123718]
See [diffoscope for ruby2.7.3-racc](https://r13y.com/diff/6fc10a25dd266369b6c7455cff325e8e38bf03dc34aff292857f7f4af5158b44-50c9d42094f69c41238e8a3e81d16d8f6cc6ccf694b4a7ec9b74a3c4ce28c7fc.html) and [diffoscope for ruby2.7.3-nokogiri](https://r13y.com/diff/214b63be4f09ab0a946177c9aed31de6777245cc7b78deaeba4a018134036f05-ffa348ffeb5947d08e19362d58be9e137e98ed13b7415f7f564288cef4034294.html).

For example, this is a difference in the file list for nokogiri (for completeness, I assume that r13y doesn't keep those logs forever):

```
# A
lib/​ruby/​gems/​2.​7.​0/​gems/​nokogiri-​1.​11.​1/​ext/​nokogiri/​.​gem.​20210503-​6-​1mgqrl9

# B
lib/​ruby/​gems/​2.​7.​0/​gems/​nokogiri-​1.​11.​1/​ext/​nokogiri/​.​gem.​20210518-​6-​1ts2gq0
```

It seems like the only thing that's preventing these from being fully reproducible is that the path contains the build date. It doesn't seem like a big challenge to change that, but I'm not familiar with the ruby nix recipes and neither with the ruby compiler / build system.

See also: https://r13y.com/

cc @raboof @alyssais

**Comment by raboof** [comment]
> See also: https://r13y.com/

Hopefully #123502 will remove those 2 from the minimal ISO closure altogether. We should still fix the reproducibility of those packages, though, of course ;).

I wonder if this is a regression - I thought asciidoctor (which seems to bring in those 2 dependencies) has been part of the minimal iso for a while, and this issue just recently popped up.

**Comment by Mindavi** [comment]
Yeah, I saw that, but these paths _seem_ like a trivial issue. Might of course be totally wrong about that, but at least it's not a giant diff that's hard to understand.

Might be a regression, haven't looked very deep into this. Just wanted to at least have a tracking issue, maybe someone has some ideas :).

**Comment by Mindavi** [comment]
I checked the release-20.09 branch and there nokogiri is also not reproducible. Also same issue with dates in the path, but also some non-determinism in ordering of some text. Seems like a minor difference though.

**Comment by Mindavi** [comment]
Closed by #123748
